src/main/java/io/anserini/search/SimpleSearcher.java

```diff
@@ -208,6 +208,48 @@ public void close() throws IOException {
     return results;
   }
 
+  // searching both the defaults contents fields and another field with weight boost
+  // this is used for MS MACRO experiments with query expansion.
+  public Result[] searchFields(String q, String f, float boost, int k) throws IOException {
+    IndexSearcher searcher = new IndexSearcher(reader);
+    searcher.setSimilarity(similarity);
+
+    Query queryContents = new BagOfWordsQueryGenerator().buildQuery(LuceneDocumentGenerator.FIELD_BODY, analyzer, q);
+    Query queryField = new BagOfWordsQueryGenerator().buildQuery(f, analyzer, q);
+    BooleanQuery query = new BooleanQuery.Builder()
+        .add(queryContents, BooleanClause.Occur.SHOULD)
+        .add(new BoostQuery(queryField, boost), BooleanClause.Occur.SHOULD).build();
+
```

Comment:
So lines below are the same with those in the function `search()`, right? Packing them into one function would be better?

Reply:
Do you mean an overloaded method `search`?

Reply:
I mean wrapping line 225 to line 250 into one function and call it in both search() and searchFields().